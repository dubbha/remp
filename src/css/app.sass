body
    margin: 0
    background-color: #ccc

.footer
    background-color: #414141
    color: #f55263
    padding: 10px
    font-family: Helvetica
    font-weight: bold
