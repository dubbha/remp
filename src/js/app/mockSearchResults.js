export default [
  {
    show_title: 'Pulp Fiction',
    category: 'Oscar-winning Movies',
    release_year: '1994',
    poster: 'http://netflixroulette.net/api/posters/880640.jpg',
    runtime: '154 min',
    summary: 'Weaving together three stories featuring a burger-loving hit man, his philosophical partner and a washed-up boxer, Quentin Tarantino influenced a generation of filmmakers with this crime caper\'s stylized, over-the-top violence and dark comic spirit.',
    director: 'Quentin Tarantino',
    show_cast: 'John Travolta, Samuel L. Jackson, Uma Thurman, Bruce Willis, Harvey Keitel, Tim Roth, Amanda Plummer, Ving Rhames, Eric Stoltz, Maria de Medeiros',
    rating: '4.1',
    show_id: 880640,
  },
  {
    show_title: 'Kill Bill: Vol. 1',
    category: 'Action & Adventure',
    release_year: '2003',
    poster: 'http://netflixroulette.net/api/posters/60031236.jpg',
    runtime: '111 min',
    summary: 'An assassin is shot by her ruthless employer, Bill, and other members of their assassination circle. But she lives -- and plots her vengeance.',
    director: 'Quentin Tarantino',
    show_cast: 'Uma Thurman, Lucy Liu, Vivica A. Fox, Daryl Hannah, David Carradine, Michael Madsen, Julie Dreyfus, Chiaki Kuriyama, Sonny Chiba, Gordon Liu',
    rating: '3.8',
    show_id: 60031236,
  },
  {
    show_title: 'Kill Bill: Vol. 2',
    category: 'Action & Adventure',
    release_year: '2004',
    poster: 'http://netflixroulette.net/api/posters/60032563.jpg',
    runtime: '137 min',
    summary: 'The Bride has three left on her rampage list: Budd, Elle Driver and Bill himself. But when she arrives at Bill\'s house, she\'s in for a surprise.',
    director: 'Quentin Tarantino',
    show_cast: 'Uma Thurman, David Carradine, Michael Madsen, Daryl Hannah, Gordon Liu, Michael Parks, Perla Haney-Jardine, Helen Kim, Claire Smithies, Clark Middleton',
    rating: '3.8',
    show_id: 60032563,
  },
  {
    show_title: 'Jackie Brown',
    category: 'Dramas',
    release_year: '1997',
    poster: 'http://netflixroulette.net/api/posters/60010514.jpg',
    runtime: '154 min',
    summary: 'Jackie Brown is an aging flight attendant who smuggles cash on the side. But when she\'s busted and pressured to help with an investigation, she plans to play the opposing forces against each other and walk away with the dough.',
    director: 'Quentin Tarantino',
    show_cast: 'Pam Grier, Samuel L. Jackson, Robert Forster, Bridget Fonda, Michael Keaton, Robert De Niro, Michael Bowen, Chris Tucker, Lisa Gay Hamilton, Tommy \'Tiny\' Lister',
    rating: '3.7',
    show_id: 60010514,
  },
  {
    show_title: 'Reservoir Dogs',
    category: 'Independent Movies',
    release_year: '1992',
    poster: 'http://netflixroulette.net/api/posters/902003.jpg',
    runtime: '99 min',
    summary: 'Quentin Tarantino\'s directorial debut is raw, violent, often mimicked -- and unforgettable. A botched robbery indicates a police informant, and the pressure mounts in the aftermath at a warehouse. Crime begets violence as the survivors unravel.',
    director: 'Quentin Tarantino',
    show_cast: 'Harvey Keitel, Tim Roth, Michael Madsen, Steve Buscemi, Chris Penn, Lawrence Tierney, Edward Bunker, Quentin Tarantino, Randy Brooks, Kirk Baltz',
    rating: '4.0',
    show_id: 902003,
  },
  {
    show_title: 'Four Rooms',
    category: 'Comedies',
    release_year: '1995',
    poster: 'http://netflixroulette.net/api/posters/520179.jpg',
    runtime: '98 min',
    summary: 'Weaving together three stories featuring a burger-loving hit man, his philosophical partner and a washed-up boxer, Quentin Tarantino influenced a generation of filmmakers with this crime caper\'s stylized, over-the-top violence and dark comic spirit.',
    director: 'Quentin Tarantino',
    show_cast: 'John Travolta, Samuel L. Jackson, Uma Thurman, Bruce Willis, Harvey Keitel, Tim Roth, Amanda Plummer, Ving Rhames, Eric Stoltz, Maria de Medeiros',
    rating: '3.6',
    show_id: 520179,
  },
  {
    show_title: 'Django Unchained',
    category: 'Dramas',
    release_year: '2012',
    poster: 'http://netflixroulette.net/api/posters/70230640.jpg',
    runtime: '154 min',
    summary: 'Accompanied by a German bounty hunter, a freed slave named Django travels across America to free his wife from a sadistic plantation owner.',
    director: 'Quentin Tarantino',
    show_cast: 'Jamie Foxx, Christoph Waltz, Leonardo DiCaprio, Kerry Washington, Samuel L. Jackson, Walton Goggins, Dennis Christopher, James Remar, Michael Parks, Don Johnson',
    rating: '4.0',
    show_id: 70230640,
  },
];
