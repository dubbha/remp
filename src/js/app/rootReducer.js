import { combineReducers } from 'redux';
import common from '../common/store/reducer';

export default combineReducers({
  common,
});
