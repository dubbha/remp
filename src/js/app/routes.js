import Search from '../search';
import Film from '../film';

export default [
  {
    path: '/search/:query?',
    component: Search,
  },
  {
    path: '/film/:id/:title?',
    component: Film,
  },
  {
    path: '/',
    exact: true,
    component: Search,
  },
];
