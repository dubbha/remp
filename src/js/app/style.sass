.app                     // IE11 flexbox min-height fix
    display: flex        // https://github.com/philipwalton/flexbugs#3-min-height-on-a-flex-container-wont-apply-to-its-flex-items
    flex-direction: row  // https://codepen.io/chriswrightdesign/pen/emQNGZ/

    &__container
        width: 100%
        min-height: 100vh
        min-width: 310px
        display: flex
        flex-direction: column
        flex-wrap: nowrap
        box-sizing: border-box
