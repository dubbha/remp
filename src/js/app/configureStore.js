import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { devToolsEnhancer } from 'redux-devtools-extension/logOnlyInProduction';
import rootReducer from './rootReducer';

export default (initialState = {}) => {
  const store = createStore(
    rootReducer,
    initialState,
    compose(
      applyMiddleware(thunk),
      devToolsEnhancer(),
    ),
  );

  return store;
};
