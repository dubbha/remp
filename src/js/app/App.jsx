import React from 'react';
import { Switch } from 'react-router-dom';
import { renderRoutes } from 'react-router-config';
import routes from './routes';
import './style.sass';

const App = () => (
  <div className="app">
    <Switch>
      {renderRoutes(routes)}
    </Switch>
  </div>
);

export default App;
