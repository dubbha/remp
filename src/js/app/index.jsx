import React from 'react';
import { render } from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { AppContainer } from 'react-hot-loader';
import { Provider } from 'react-redux';
import configureStore from './configureStore';
import App from './App';
import moduleStub from './moduleStub';

const store = configureStore();

const renderApp = () => {
  render(
    <AppContainer>
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    </AppContainer>,
    document.getElementById('app'),
  );
};

renderApp();

moduleStub(module); // we only need this stub to be able to test the code block below
if (module.hot) {
  module.hot.accept('./App', renderApp);
}
