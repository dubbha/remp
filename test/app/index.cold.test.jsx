import React from 'react';

jest.mock('react-hot-loader', () => ({
  AppContainer({ children }) {
    return (<x-app-container>{children}</x-app-container>);
  },
}));

jest.mock('react-redux', () => ({
  Provider({ children }) {
    return (<x-provider>{children}</x-provider>);
  },
}));

jest.mock('react-router-dom', () => ({
  BrowserRouter({ children }) {
    return (<x-browser-router>{children}</x-browser-router>);
  },
}));

jest.mock('app/App', () => function App() { return <x-app />; });

jest.mock('app/configureStore', () => jest.fn());

describe('app/index', () => {
  it('should render successfully', () => {
    require('app');
    expect(document.getElementById().innerHTML).toMatchSnapshot();
  });
});

