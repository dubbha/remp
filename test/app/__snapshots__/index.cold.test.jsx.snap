// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/index should render successfully 1`] = `"<x-app-container><x-provider><x-browser-router><x-app></x-app></x-browser-router></x-provider></x-app-container>"`;
